rewards/classes/TreeManager.py

```diff
@@ -52,122 +55,45 @@ def convert_to_merkle_tree(self, rewardsList: RewardsList, start: int, end: int)
 
     def build_function_and_send(self, account, func) -> str:
         options = self.get_tx_options(account)
-        console.log(options)
         tx = func.buildTransaction(options)
         signed_tx = self.w3.eth.account.sign_transaction(tx, private_key=account.key)
         tx_hash = self.w3.eth.send_raw_transaction(signed_tx.rawTransaction).hex()
         return tx_hash
 
-    def approve_root(self, rewards) -> str:
+    def approve_root(self, rewards) -> Tuple[str, bool]:
         console.log("Approving root")
-        approve_root_func = self.badgerTree.approveRoot(
-            to_bytes(hexstr=rewards["merkleTree"]["merkleRoot"]),
-            to_bytes(hexstr=rewards["rootHash"]),
+        return self.manage_root(rewards, self.badgerTree.approveRoot, approve=True)
+
+    def propose_root(self, rewards):
+        console.log("Proposing root")
+        return self.manage_root(rewards, self.badgerTree.proposeRoot, approve=False)
+
+    def manage_root(self, rewards, contract_function, approve):
```

Comment:
can you add type hints for the params, and maybe rename approve action, make it a str and let it be "approve" or "propose"? feel like that's a little more intuitive, if you like current approach better though that's ok with me